compileopts/target.go

```diff
@@ -38,6 +38,7 @@ type TargetSpec struct {
 	AutoStackSize    *bool    `json:"automatic-stack-size"` // Determine stack size automatically at compile time.
 	DefaultStackSize uint64   `json:"default-stack-size"`   // Default stack size if the size couldn't be determined at compile time.
 	CFlags           []string `json:"cflags"`
+	CXXFlags         []string `json:"cxxflags"`
```

Comment:
I don't see why a `cxxflags` key is necessary?
C flags are important in the target file because they define things like the float ABI. But these flags are also used for C++. I can't think of a reason why you would want to configure C++ flags in a target file.

Reply:
`cxxflags` is needed for compiling C++ source code. If you move cxx flags such as `-std=c++17` to `cflags`, you will get the following error when compiling the C source code.
```
error: invalid argument '-std=c++17' not allowed with 'C'
```